keras/backend/tensorflow/optimizer.py

```diff
@@ -33,6 +33,9 @@ def add_variable_from_reference(
             return super().add_variable_from_reference(
                 reference_variable, name=name, initializer=initializer
             )
+    def add_variable(self, shape, initializer="zeros", dtype=None, name=None):
+        with tf.init_scope():
```

Comment:
Does this need to special-case Optimizer? Could it be moved to `backend/tensorflow/core.py`?

Reply:
In general, moving it to core will be ideal. We only hit this in optimizer since the optimizer wights are usually created when tracing train function (not same as the model weights which are created eagerly.)

Reply:
Sending https://github.com/keras-team/keras/pull/19011 for now.